app/assets/javascripts/Patient.js

```diff
@@ -0,0 +1,143 @@
+  return this.get_history_elements('care_goal');
+}
+
+PatientSchema.methods.conditions = function conditions(params, callback) {
+  return this.get_history_elements('condition');
+}
+
+PatientSchema.methods.encounters = function encounters(params, callback) {
+  return this.get_history_elements('encounter');
+}
+
+PatientSchema.methods.communications = function communications(params, callback) {
+  return this.get_history_elements('communication');
+}
+
+PatientSchema.methods.family_history = function family_history(params, callback) {
+  return this.get_history_elements('family_history');
+}
+
+PatientSchema.methods.immunizations = function immunizations(params, callback) {
+  return this.get_history_elements('immunization');
+}
+
+PatientSchema.methods.medical_equipment = function medical_equipment(params, callback) {
+  return this.get_history_elements('medical_equipment');
+}
+
+PatientSchema.methods.medications = function medications(params, callback) {
+  return this.get_history_elements('medication');
+}
+
+PatientSchema.methods.procedures = function procedures(params, callback) {
+  return this.get_history_elements('procedure');
+}
+
+PatientSchema.methods.results = function results(params, callback) {
+  return this.get_history_elements('result');
+}
+
+PatientSchema.methods.social_history = function social_history(params, callback) {
+  return this.get_history_elements('social_history');
+}
+
+PatientSchema.methods.vital_signs = function vital_signs(params, callback) {
+  return this.get_history_elements('vital_sign');
+}
+
+PatientSchema.methods.devices = function devices(params, callback) {
+  return this.get_history_elements('device');
+}
+
+PatientSchema.methods.diagnostic_studies = function diagnostic_studies(params, callback) {
+  return this.get_history_elements('diagnostic_study');
+}
+
+PatientSchema.methods.functional_statuses = function functional_statuses(params, callback) {
+  return this.get_history_elements('functional_status');
+}
+
+PatientSchema.methods.interventions = function interventions(params, callback) {
+  return this.get_history_elements('intervention');
+}
+
+PatientSchema.methods.laboratory_tests = function laboratory_tests(params, callback) {
+  return this.get_history_elements('laboratory_test');
+}
+
+PatientSchema.methods.physical_exams = function physical_exams(params, callback) {
+  return this.get_history_elements('physical_exam');
+}
+
+PatientSchema.methods.risk_category_assessments = function risk_category_assessments(params, callback) {
+  return this.get_history_elements('risk_category_assessment');
+}
+
+PatientSchema.methods.care_experiences = function care_experiences(params, callback) {
+  return this.get_history_elements('care_experience');
+}
+
+PatientSchema.methods.preferences = function preferences(params, callback) {
+  return this.get_history_elements('preference');
+}
+
+PatientSchema.methods.provider_characteristics = function provider_characteristics(params, callback) {
```

Comment:
Do we need to add a function for 'patient_characteristic' which is the category for the 'payer' model?

Reply:
All of the patient characteristics (minus expired, which is a condition) are fields on the patient itself, rather than entries in the `history_elements`.

That being said, those characteristics _ARE_ still models coming from the modelinfo file, which theoretically could at some point be entries in `history_elements`, so I think it does make sense to have a helper method for those types specifically. I will add that in!

Reply:
i'm going to forward an email thread to you about patient characteristics. it sounds like measure developers can (and are interested in) adding random things in as patient characteristics. I don't think it makes sense to add here, but I'll send to you so you can take a look at it.

Reply:
There is a generic `patient_characteristic` model in the spec, with just one attribute (`author_datetime`). We could add another field to it (maybe `data`?) that could hold anything (Code, Quantity, Interval, etc.), and any time measure developers want to add whatever, they could use it.

Thoughts?